applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py

```diff
@@ -90,6 +94,11 @@ def UpdateTimeStep(self, step):
     def UpdateMeshAccordingInputVariable(self, InputVariable):
         self.mesh_controller.UpdateMeshAccordingInputVariable(InputVariable)
 
+        if self.model_settings["damping"]["recalculate_damping"].GetBool():
```

Comment:
In case you recalculate the weights, you do not initialize a damping_utility in the initialize function. Instead, you initialize it here AFTER the mesh is updated. That is very confusing and only makes sense because in the first time "UpdateMeshAccordingInputVariable" is called in the opt algorithms, the shape update is zero. I would initiate self.damping_utility in any case in the initialize function and after the mesh is updated, it also gets updated. Imagine you create a model part controller and want to damp a variable using "DampNodalVariableIfSpecified" without having called "UpdateMeshAccordingInputVariable", which is a fair assumption as otherwise, the functions would imply a dedicated call sequence.

Or am I overseeing here something?

Reply:
Good point, only that in first iteration the damping weights would be calculated twice then. 

An alternative approach would be to initialize right before damping in case the utils are None, and set them to None when the mesh is updated and the damping should be always recalculated.